skimage/restoration/tests/test_rolling_ball.py

```diff
@@ -18,6 +19,15 @@ def test_ellipsoid_const():
     assert np.allclose(img - background, np.zeros_like(img))
 
 
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_rolling_ball_dtype(dtype):
+    img = 155 * np.ones((100, 100), dtype=dtype)
+    kernel = ellipsoid_kernel((25, 53), 50).astype(dtype, copy=False)
+    background = rolling_ball(img, kernel=kernel)
+    assert background.dtype == img.dtype
```

Comment:
This will always be true, because we cast `background` to `img.dtype` after applying the kernel. The question is if we want that or if it should be `background.dtype == np.float32` if img is F16.

Reply:
right, thanks!

Reply:
I have left the dtype behavior in rolling_ball as-is for now. I think this was mainly as a convenience for users providing integer valued inputs to get integer valued outputs?

Reply:
Yes. We introduced this because of integer images - I never wrote an optimized integer kernel -, but it affects all inputs equally. The general idea was "dtype in, dtype out".

I think it will be best if the behavior here regarding the return type is consistent with the other functions in the library. If that means changing the current behavior, I think it makes sense to do.

Reply:
I think this concern is orthogonal to the purpose if this PR, so let's just defer this decision so as not to hold the PR up unnecessarily.